src/bjointsp/read_write/writer.py

```diff
@@ -13,7 +13,7 @@ def create_result_file(input_files, subfolder, seed=None, seed_subfolder=False,
     file_name = ""
     # add basename of each input file to the output filename
     for f in input_files:
-        if f is not None:
+        if f is not None and isinstance(f, str):
```

Comment:
What's that for?

Reply:
So when originally BJointSP gets source and template file paths they are strings. But when we pass objects, template is a dict and source is a list. So I would only want to add the network name to file-name in case we send objects and we are writing results.

Reply:
just add a comment there too